app/controllers/exam_controller.py

```diff
@@ -0,0 +1,146 @@
+import json
+from app.models.exam import Exam
+from app.models.user import User
+from app import response, app, db
+from flask import request, jsonify
+from flask_jwt_extended import *
+import datetime
+
+def format_exam(exam):
+    return {
+        "id": exam.id,
+        "question": exam.question,
+        "correct_option": exam.correct_option,
+        "option_a": exam.option_a,
+        "option_b": exam.option_b,
+        "option_c": exam.option_c,
+        "option_d": exam.option_d,
+        "user_id": exam.user_id,
+        "material_id" : exam.material_id,
+        "created_at": exam.created_at,
+        "updated_at": exam.updated_at
+    }
+
+def get_all_exam():
+    try:
+        exams = Exam.query.all()
+        formated_exams = [format_exam(exam) for exam in exams]
+        return jsonify({"exams": formated_exams})
+    except Exception as e:
+        return jsonify({"error": f"An error occurred: {e}"}), 500
+
+def get_exam_by_id(id):
+    try:
+        GetExam = Exam.query.get(id)
+        if GetExam :
+            exam = {
+                 "id": GetExam.id,
+                "question": GetExam.question,
+                "correct_option": GetExam.correct_option,
+                "option_a": GetExam.option_a,
+                "option_b": GetExam.option_b,
+                "option_c": GetExam.option_c,
+                "option_d": GetExam.option_d,
+                "user_id": GetExam.user_id,
+                "material_id" : GetExam.material_id,
+                "created_at": GetExam.created_at,
+                "updated_at": GetExam.updated_at
+            }
+            return jsonify(exam)
```

Comment:
is there a reason not to return jsonify(object) directly without moving its content to a new dictionary?

let's say you update GetExam with exam. Can you just return jsonify exam directly?

```if exam:
   return jsonify(exam)```

Reply:
done